### ISSUE TITLE
Shape of the `data` array

### ISSUE THREAD

**Issue by pbugnion** [pbugnion/gmaps#68]
At the moment, layers accept a data array that is a list of (_latitude_, _longitude_) pairs. An alternative interface would be to have a pair of _latitude_, _longitude_ arrays.

The motivation for passing in two arrays, one with _latitude_ and one with _longitude_ values, is that most modellers, data scientists etc. are used to dealing with tabular data, e.g. through a pandas dataframe. They might have a column of latitudes and a column of longitudes in their data frame.

Given that this is a major breaking change in the API if implemented, it's important to resolve this as early as possible.

**Comment by pbugnion** [comment]
I'm not convinced that having two separate arrays makes it really that much easier. @reyale, you say that it would allow this interface:

```python
import pandas as pd
df = pd.read_csv('data.csv')
result = df[(df['lat'] > 40) & (df['lon'] > 40)]
Layer(data=[result['lat'],result['lon']])
```

However, the current interface allows very similar behaviour:

```python
import pandas as pd
df = pd.read_csv('data.csv')
result = df[(df['lat'] > 40) & (df['lon'] > 40)]
Layer(data=[result[['lat', 'lon']])
```
... which seems just as easy.

Similarly, if the data is in two lists:

```python
Heatmap(data=zip(latitudes, longitudes))
```
... is very easy to use.

Ultimately, what makes the array of *latitude*, *longitude* pairs most appealing to me is that a pair is really a single object. A latitude is meaningless without a longitude (in terms of expressing the location of a point on the globe). I think of the `data` array as a single array of points, rather than an array of pairs of floats. 

I think that we are really optimising for two different things. The proposed new interface may be superior for someone dealing entirely with tabular data. The current interface optimises for conceptual simplicity.

**Comment by reyale** [comment]
`Layer(data=[result[['lat', 'lon']])`

This doesn't work, at least with pandas '0.18.1' and gmaps '0.3.0'.  This is because result[['lat', 'lon'] returns a dataframe of two columns, not a vector of lat-long pairs.  This transformation between two columns and one list of tuples is the common transformation that I am suggesting should be moved to the library.

`Heatmap(data=zip(latitudes, longitudes))`

Will work, for reference.

> pair is really a single object

I have sympathy for this view, but it's not how data scientists operate, and it's not how data is stored.  

>I think that we are really optimising for two different things. The proposed new interface may be superior for someone dealing entirely with tabular data. The current interface optimises for conceptual simplicity.

I think you're a bit hung up on how the interface presents versus what operation should be moved in to the library.  If the operation of zipping vectors is required almost every time someone uses `Layer(data=...)` the zip operation belongs in the library itself.  If you're really hung up on making the lat-lon pairs interface available you can provide support for both data input styles using a number of python paradigms (kwargs, dtype runtime detection, ect).  Something as simple as `type(t[0]) == tuple` will allow you detect how the data is presented.  

There's another added benefit here in that if the zip operation ever becomes the bottleneck  of the api having your zip operation in the library gives you control of its optimization in the future.

**Comment by pbugnion** [comment]
Closed by PR #91
